### ISSUE TITLE
Strange compiled asset import issue

### ISSUE THREAD

**Issue by wavded** [oven-sh/bun#3006]
### What version of Bun is running?

0.6.3

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

This is really strange one:

Create the following tree:

```
test/test.ts
index.ts
```

In `test.ts` put the following:

```
class Test {}
export default Test
```

In `index.ts` put the following:

```
import Test from "./test/test.ts"
new Test()
```

Run `bun index.ts`. No output, no errors. 👍🏻 

Run `bun build index.ts --compile --outfile app` then run `./app`. Get an error: 👎🏻 

```
23 | });
24 |
25 | // index.ts
26 | var test = __toESM(require_test());
27 |
28 | new test.default;
   ^
TypeError: Object is not a constructor (evaluating 'new test.default')
```

**Change the name of the folder from `test` to anything else and it works without issues.** 😱 

### What is the expected behavior?

Running `bun index.ts` and the compiled `./app` should produce the same result.

### What do you see instead?

The compiled app seems to have an issue with the import.

### Additional information

Even weirder. If you change `test.ts` to use a named export instead of the default, you can keep the folder named `test` and it will work:

Update `test.ts` to:

```
export class Test {}
export default Test
```

And `index.ts` to:

```
import {Test} from "./test/test.ts"
new Test()
```

The issue goes away.

**Comment by Jarred-Sumner** [comment]
The bug is that Bun is erroneously converting `test/test.ts` into a CommonJS file. For some reason, it's "losing" the `export default` there.

**Comment by Electroid** [comment]
This has been fixed as of Bun v1.0.26.
